include/syntax/swift

```diff
@@ -0,0 +1,412 @@
+// supported feature lists -----------------------------------------------------
+
+supported_apis = ["simple", "generic", "record"];
+supported_api_styles = ["free-form"];
+supported_code_models = ["loop-switch", "recursive-functions"];
+supported_targets = ["code", "dot"];
+supported_features = ["nested-ifs", "bitmaps", "case-ranges", "tags", "captures",
+    "captvars"];
+
+
+// language-specific options ---------------------------------------------------
+
+semicolons = 0;
+backtick_quoted_strings = 0;
+single_quoted_strings = 0;
+indentation_sensitive = 0;
+wrap_blocks_in_braces = 0;
+special_escapes = "\a\b\t\n\v\f\r\"\\";
```

Comment:
Are these all really supported in Swift? As far as I read here https://docs.swift.org/swift-book/documentation/the-swift-programming-language/stringsandcharacters you don't need `\a, \b, \f, \v` (`\0` can be ignored):

> The escaped special characters \0 (null character), \\ (backslash), \t (horizontal tab), \n (line feed), \r (carriage return), \" (double quotation mark) and \' (single quotation mark)

I guess it doesn't show up on the test because you are using hex literals anyway.

Reply:
They also don't show up in the [document](https://docs.swift.org/swift-book/documentation/the-swift-programming-language/lexicalstructure/#String-Literals) I was referencing too, I think I just misunderstood the purpose of this configuration (looking at the test I definitely recognised that they didn't exist in the language), I also missed the single quote, probably because I didn't think it would ever need escaping since they're unused.

I've committed something that should hopefully fix this.

Reply:
The purpose of this configuration is for re2c to know if certain characters need to be rendered in a special way. When it generates the matching clauses, it emits characters like `'a'` or `\xff`, but special escape symbols need custom rendering rules. For swift this is irrelevant at the moment, since characters are rendered as hex numbers.